File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(string my_string) {
    int answer = 0;
    string temp = "";
    
    for(char c : my_string) {
        if(c >= '0' && c <= '9') {
            temp += c;
        } else {
            if(atoi(temp.c_str())) {
                answer += stoi(temp);
                temp = "";
            }
        }
    }
    if(!temp.empty()) answer += stoi(temp);
    return answer;
}

File: 프로그래머스/0/120864. 숨어있는 숫자의 덧셈 （2）/숨어있는 숫자의 덧셈 （2）.swift
import Foundation

func solution(_ my_string:String) -> Int {
    var result = 0
    var str = ""
    for i in my_string {
        if i.isNumber {
            str += String(i)
        } else {
            result += Int(str) ?? 0
            str = ""
        }
    }
    result += Int(str) ?? 0
    return result
}